packages/@react-spectrum/card/src/CardBase.tsx

```diff
@@ -69,13 +69,16 @@ function CardBase<T extends object>(props: CardBaseProps<T>, ref: DOMRef<HTMLDiv
 
   // this is for horizontal cards
   let [height, setHeight] = useState(NaN);
-  useLayoutEffect(() => {
+  let updateHeight = useCallback(() => {
     if (orientation !== 'horizontal') {
       return;
     }
+
     let cardHeight = gridRef.current.getBoundingClientRect().height;
     setHeight(cardHeight);
-  }, [gridRef, setHeight, orientation]);
+  }, [orientation, gridRef, setHeight]);
+  useResizeObserver({ref: gridRef, onResize: updateHeight});
+
```

Comment:
The horizontal card images weren't getting their heights properly updated when resizing the CardView div so moved to `useResizeObserver`

Reply:
Is a resize observer on all cards expensive?

Reply:
Yeah, ideally we'd use the `aspect-ratio` css attribute as a pure css solution and get rid of this but it has pretty iffy/buggy performance on Firefox and was only relatively recently added to Safari so we have browser compatibility to worry about. I couldn't think of any alternatives to the resize observer, happy to field any suggestions

Reply:
Well, perhaps something to follow up on. Perhaps the CardView could trigger this somehow since it already has a resize observer. It's unlikely that one card would change dimensions when the card view did not.